### ISSUE TITLE
Add text indentation support at paragraph block type level

### ISSUE THREAD

**Issue by kjellr** [WordPress/gutenberg#37462]
Currently, paragraphs in the editor have vertical space to differentiate one paragraph from the next: 

<img width="375" alt="Screen Shot 2021-12-16 at 12 17 34 PM" src="https://user-images.githubusercontent.com/1202812/146418412-40199a63-ba0d-4182-b599-5fead7e9ed95.png">

To allow for an optional, more "literary" treatment, we should consider adding an option to Global Styles that would collapse that margin and instead use indentation to differentiate from one paragraph to the next: 

<img width="375" alt="Screen Shot 2021-12-16 at 12 18 10 PM" src="https://user-images.githubusercontent.com/1202812/146418475-e0fadec5-96f4-4d8f-8015-f323ae8fa9ce.png">

---

Similar: https://github.com/WordPress/gutenberg/issues/12451

**Comment by mtias** [comment]
Yes! I'd been wanting to do this for a while, but needed more of the global styles infrastructure in place. I also want to allow justification (with localized hyphens and such) to happen at this level globally so it can be undone easily.

It can also be an opportunity to include some other typographical flair, depending on what the fonts support (discretionary ligatures, etc).

**Comment by skorasaurus** [comment]
related https://github.com/WordPress/gutenberg/issues/34688

**Comment by ericmacknight** [comment]
Yes, please!

**Comment by justintadlock** [comment]
Just adding a +1 here. For folks like me who have some literary or academic work and want to preserve formatting, having a way to text-indent paragraphs would be super helpful.

I've been shipping an `indent` block style in my theme work to take care of this need (in case anyone needs a solution in the meantime): https://gist.github.com/justintadlock/257087c52e46272f1445fd8ff60c8d83

**Comment by t-hamano** [comment]
The text indentation support has been reverted. See #74200 for the reasons why, the indentation conventions in various locales, and the issues involved in dealing with them.

**Comment by aaronrobertshaw** [comment]
A draft PR, https://github.com/WordPress/gutenberg/pull/74863,  is up exploring a different approach that will limit line indent global styles to only the paragraph block while also offering the ability to arbitrarily decide whether the indentation applies to only subsequent or all paragraph blocks.

**Comment by aaronrobertshaw** [comment]
There's a more promising alternative approach over in https://github.com/WordPress/gutenberg/pull/74889. I've close #74863 in favour of this.
